src/libraries/Microsoft.PowerFx.Core/Lexer/TexlLexer.cs

```diff
@@ -75,316 +67,64 @@ public enum Flags
-            AddPunctuator(PunctuatorParenClose, TokKind.ParenClose);
-            AddPunctuator(PunctuatorEqual, TokKind.Equ);
-            AddPunctuator(PunctuatorLess, TokKind.Lss);
-            AddPunctuator(PunctuatorLessOrEqual, TokKind.LssEqu);
-            AddPunctuator(PunctuatorGreater, TokKind.Grt);
-            AddPunctuator(PunctuatorGreaterOrEqual, TokKind.GrtEqu);
-            AddPunctuator(PunctuatorNotEqual, TokKind.LssGrt);
-            AddPunctuator(LocalizedPunctuatorListSeparator, TokKind.Comma);
-            AddPunctuator(PunctuatorDot, TokKind.Dot);
-            AddPunctuator(PunctuatorColon, TokKind.Colon);
-            AddPunctuator(PunctuatorCurlyOpen, TokKind.CurlyOpen);
-            AddPunctuator(PunctuatorCurlyClose, TokKind.CurlyClose);
-            AddPunctuator(PunctuatorBracketOpen, TokKind.BracketOpen);
-            AddPunctuator(PunctuatorBracketClose, TokKind.BracketClose);
-            AddPunctuator(PunctuatorAmpersand, TokKind.Ampersand);
-            AddPunctuator(PunctuatorPercent, TokKind.PercentSign);
-            AddPunctuator(LocalizedPunctuatorChainingSeparator, TokKind.Semicolon);
-            AddPunctuator(PunctuatorAt, TokKind.At);
-
-            //Commenting punctuators
-            AddPunctuator(PunctuatorBlockComment, TokKind.Comment);
-            AddPunctuator(PunctuatorLineComment, TokKind.Comment);
-
-            // Keywords
-            AddKeyword(KeywordTrue, TokKind.True);
-            AddKeyword(KeywordFalse, TokKind.False);
-            AddKeyword(KeywordIn, TokKind.In);
-            AddKeyword(KeywordExactin, TokKind.Exactin);
-            AddKeyword(KeywordSelf, TokKind.Self);
-            AddKeyword(KeywordParent, TokKind.Parent);
-            AddKeyword(KeywordAnd, TokKind.KeyAnd);
-            AddKeyword(KeywordOr, TokKind.KeyOr);
-            AddKeyword(KeywordNot, TokKind.KeyNot);
-            AddKeyword(KeywordAs, TokKind.As);
-
-            PopulateKeywordArrays();
-        }
+        // Limits the StringBuilderCache TLS memory usage for LexerImpl.
+        // Usually our tokens are less than 128 characters long, unless it's a large string.
+        private const int DesiredStringBuilderSize = 128;
 
-        // This creates a new Lexer with the same locale as the original, which is a lot less expensive than spinning up a whole new one
-        private TexlLexer(TexlLexer original)
-        {
-            Contracts.AssertValue(original);
+        public static TexlLexer InvariantLexer { get; } = new TexlLexer(PunctuatorDecimalSeparatorInvariant);
 
-            LocalizedPunctuatorDecimalSeparator = original.LocalizedPunctuatorDecimalSeparator;
-            LocalizedPunctuatorListSeparator = original.LocalizedPunctuatorListSeparator;
-            LocalizedPunctuatorChainingSeparator = original.LocalizedPunctuatorChainingSeparator;
+        public static TexlLexer CommaDecimalSeparatorLexer { get; } = new TexlLexer(PunctuatorCommaInvariant);
```

Comment:
IT looks like this is one of the key simplifications? That we really only have 2 lexers, and so we don't need a CultureInfo-->Lexer cache.

Reply:
Yes, this is one of the key pieces. We only actually ever have two lexers